Dockerfile

```diff
@@ -1,23 +1,47 @@
-FROM ubuntu:20.04
+# --------------------------------------------------
+# Base container
+# --------------------------------------------------
+FROM docker.io/ubuntu:focal AS base
 
 RUN set -e
 
-# Setup time-zone so that the build does not hang
-# on configuring the tzdata package.
-# I work in Brno, that is basically Vienna-North :)
-# (definitely its closer than Prague)
-ENV TZ=Europe/Vienna
-RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime
-RUN echo "$TZ" > /etc/timezone
-
 # Install packages
+ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update
-RUN apt-get install -y git cmake make llvm zlib1g-dev clang g++ python3
+RUN apt-get install -yq --no-install-recommends clang emacs llvm vim
+
+# --------------------------------------------------
+# Build container
+# --------------------------------------------------
+FROM base as build
 
+# Install build dependencies
+RUN apt-get install -yq --no-install-recommends ca-certificates cmake g++ git \
+                                                ninja-build llvm-dev python3
+
+# Clone
 WORKDIR /opt
 RUN git clone https://github.com/mchalupa/dg
+
+# Build
 WORKDIR /opt/dg
-RUN cmake . -DCMAKE_INSTALL_PREFIX=/usr
-RUN make -j2
-RUN make check -j2
-RUN make install
+RUN cmake -S. -GNinja -Bbuild -DCMAKE_INSTALL_PREFIX=/dg
+RUN cmake --build build
+RUN cmake --build build --target check
+
+# Install
+RUN cmake --install build
+
+# -------------------------------------------------
+# Release container
+# -------------------------------------------------
+FROM base AS release
+
+# Podman does merging with COPY --from=build /dg / but Docker just overwrites
+# everything in / ...
+COPY --from=build /dg/bin /bin
+COPY --from=build /dg/include /usr/include
+COPY --from=build /dg/lib /lib
```

Comment:
Wouldn't it be easier and more robust just to set `PATH` and `LD_LIBRARY_PATH`? (I do not know)

Reply:
Yeah, that could work as well.